app.rb

```diff
@@ -22,13 +22,20 @@ class App < Sinatra::Base
   post '/gateway' do
     incoming_message = params['Body'].to_s.downcase
     cmd, arg_text = incoming_message.split(/\s+/, 2)
+    sender_whitelist = ENV['SENDER_WHITELIST'] ? ENV['SENDER_WHITELIST'].gsub('-', '').split(',') : nil
```

Comment:
Can this get pulled out into an object so we can test it easier?

```ruby
# lib/phone_whitelist.rb
whitelist = PhoneWhitelist.new(ENV["SENDER_WHITELIST"])

if whitelist.valid_number?(params['From')
  # ... send msg
else
  # log error
end
```

And write a few tests for the different formats that you have in the README? This will keep this main `post /gateway` handler lighter.

Reply:
Great idea - can do! I'll be looking into how to write the tests.

Reply:
As far as the tests go, you can create a file like:

```ruby
# spec/lib/phone_whitelist_spec.rb
require_relative "../../lib/phone_whitelist.rb"

describe PhoneWhitelist do
  describe "#valid_number?" do
    # ... write your tests
  end
end
```

You can just make spec examples like:

```ruby
whitelist = PhoneWhitelist.new("+12061234567")

expect(whitelist.valid_number?("2061234567")).to eq(true)
```

or whatever behavior you are trying to test. Thanks!

Reply:
See: https://rspec.info as well as the existing `spec/` dir

Reply:
Awesome, thanks for the tips!